### PR TITLE
Add test for regressions on check update

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Adds a few tests to ensure regressions produce the regressions summary, and no regressions does not.

### CODE REVIEW

File: shared/run_diff.go [hunk 1]
```diff
@@ -150,15 +150,18 @@ func NewTestDiff(before, after []int, filter DiffFilterParam) TestDiff {
 	}
 }
 
+// ResultsDiff is a collection of test diffs, mapped by the test path.
```
Comment (LukeZielinski) [inline shared/run_diff.go]: nit: mapped->keyed?

File: api/checks/update_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/web-platform-tests/wpt.fyi/api/checks/summaries"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"github.com/web-platform-tests/wpt.fyi/shared/sharedtest"
+)
+
+func TestGetDiffSummary_Regressed(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	before := shared.TestRun{}
```
Comment (LukeZielinski) [inline api/checks/update_test.go]: Can you add helper functions for some of the common code? Might make it easier to tell what's changing between tests. AFAICT it's just the TestDiff and the summary lines, right?

File: api/checks/update_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	before.FullRevisionHash = strings.Repeat("0", 40)
+	before.BrowserName = "chrome"
+	after := shared.TestRun{}
+	after.FullRevisionHash = strings.Repeat("1", 40)
+	after.BrowserName = "chrome"
+	runDiff := shared.RunDiff{
+		Differences: shared.ResultsDiff{"/foo.html": shared.TestDiff{0, 1, 0}},
```
Comment (LukeZielinski) [inline api/checks/update_test.go]: nit: probably can't hurt to comment why a TestDiff of 0,1,0 is a regression while 1,0,1 is a completion

File: shared/run_diff_test.go [hunk 4]
```diff
@@ -158,16 +158,19 @@ func getDeltaResultsMaps(before []int, after []int) (map[string][]int, map[strin
...
+		"/foo.html": TestDiff{0, 1, 0},
 	}
 	regressions := diff.Regressions()
 	assert.Equal(t, 1, regressions.Cardinality())
 	assert.True(t, regressions.Contains("/foo.html"))
 
-	diff.Differences["/foo.html"] = TestDiff{1, 0, 1}
+	diff = ResultsDiff{"/bar.html": TestDiff{1, 0, 1}}
 	regressions = diff.Regressions()
 	assert.Equal(t, 0, regressions.Cardinality())
```
Comment (LukeZielinski) [inline shared/run_diff_test.go]: also assert that regressions does not contain bar.html?

File: shared/run_diff.go (hunk 1 above)
Comment (lukebjerring) [reply shared/run_diff.go]: Done.

File: api/checks/update_test.go (hunk 3 above)
Comment (lukebjerring) [reply api/checks/update_test.go]: Done via a little overkill assertion.

File: shared/run_diff_test.go (hunk 4 above)
Comment (lukebjerring) [reply shared/run_diff_test.go]: Done.

**Pull request merged**